hazelcast/src/main/java/com/hazelcast/datalink/ReferenceCounter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datalink;
+
+import com.hazelcast.core.HazelcastException;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class ReferenceCounter {
+
+    private final AtomicInteger referenceCount = new AtomicInteger(1);
+    private final Callable<Void> destroy;
```

Comment:
Callable -> RunnableEx?

This way we don't have to add `return null` (which is not used in destroy method) and we can have exception-throwing code

Reply:
5ef5dea9396